### PR TITLE
Update contributor templates [2]

### PR DESCRIPTION
**Pull request by Songtech-0912**
## PR Description

image

This PR aims to do these things:

* Update the PR template to a more intuitive and complete form, with set fields for content and checklists to follow
* Split the issues template to 2 separate templates, consistent with most other open-source projects:
   * The `feature.yml` template is a form-based template that leverages the [newest issue form system](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms) to make both managing and creating new issues easier and more standardized
   * The `bugs.yml` template is also form-based and adds more accurate fields for filling out a bug report
* All templates have been extensively tested via Github's web-based interface

View a [live demo here](https://github.com/Songtech-0912/Natron/issues/new/choose).

Some screenshots of the improvements:

image

(Feature and bug report are still subject to change so they aren't shown here)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    validations:
+      required: true
+  # System (hardware) info
+  - type: textarea
+    attributes:
+      label: System specs
+      render: markdown
+      placeholder: |
+        RAM: 15.6 GiB
+        CPU: 8 x Intel Core i7-7700HQ CPU @ 2.80GHz
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/bug.yml]: This is a confusing syntax full of information that probably isn't helpful to us.  We likely do not care about clock speed, core count is easy to look up.  "Intel Core i7-7700HQ" (processor model) is likely enough for this.  People will confuse what you've listed as core count with socket count.
```suggestion
        CPU: Intel Core i7-7700HQ
```

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        GPU: NVIDIA GeForce GTX 1070/PCle/SSE2
+      description: |
+        Please input the following (if you know them):
+          - Your system's RAM
+          - The number, model, and type of your CPU(s)
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/bug.yml]: "The number" is redundant and covered by the "s" in CPU(s).  If people have multi-socketed systems they will presumably specify that
```suggestion
          - The model of your CPU(s)
```

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+          - The model and type of your GPU
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/bug.yml]:
```suggestion
          - The model of your GPU(s)
```

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  # Did the user install via the official installer?
+  - type: checkboxes
+    attributes:
+      label: Did you install Natron using the official installer?
+      description: If you installed Natron via another method, please add your installation method in the \"additional details\" section at the bottom of this form
+      options:
+        - label: Yes, I used the official installer
+        - label: No, I installed from a binary archive
+        - label: No, I compiled Natron from sources
+        - label: No, I installed Natron via another method
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/bug.yml:70]: Issues with flatpak seem to come up a lot...  Should this be a labeled option?

**Comment by Songtech-0912** [comment]
@Shrinks99 Fixed the placeholder info as you requested. Let's work on this in Github rather than the weird switching between here and discord.

**Comment by Songtech-0912** [comment]
Also, to answer your previous question, "other - please specify" is not an option supported by the new issues form feature after searching through docs.

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    validations:
+      required: true
+  # At which location did the user install Natron?
+  - type: textarea
+    attributes:
+      label: At which location did you install Natron?
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/bug.yml]: Most people will install Natron to the default location.  I don't think this should be required and should be reworded accordingly
```suggestion
      label: If you installed to a custom (non-default) location please include it here:
```

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+      placeholder: E.g. I installed Natron at `/opt/Natron-2.4.0`
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/bug.yml]:
```suggestion
      placeholder: `/opt/Natron-2.4.0`
```

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+          required: true
+        - label: This issue is not a feature request or a pull request
+          required: true
+  # Natron & OS versions
+  - type: input
+    id: natron-version
+    attributes:
+      label: Natron version
+      description: Natron version/commit (they can be retrieved from the about window or with `natron --about`)
+      placeholder: "e.g. Natron 2.4.0"
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/bug.yml]: People will understand that this is an example placeholder text, it's a pretty common convention with forms and we don't need to specify that it's an example every time.
```suggestion
      placeholder: "Natron 2.4.0"
```

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+    validations:
+      required: true
+  - type: input
+    id: os-version
+    attributes:
+      label: Operating system
+      description: The name and version of your operating system
+      placeholder: "e.g. macOS 10.15 Catalina"
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/bug.yml]:
```suggestion
      placeholder: "macOS 10.15 Catalina"
```

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+    validations:
+      required: true
+  # At which location did the user install Natron?
+  - type: textarea
+    attributes:
+      label: At which location did you install Natron?
+      placeholder: E.g. I installed Natron at `/opt/Natron-2.4.0`
+      render: markdown
+    validations:
+      required: true
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/bug.yml]:
```suggestion
      required: false
```

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+  # What was the user trying to do?
+  - type: textarea
+    attributes:
+      label: What were you trying to do?
+      placeholder: E.g. While working on a project in Natron I tried to...
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/bug.yml]:
```suggestion
      placeholder: While working on a project in Natron I tried to...
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 11]
```diff
@@ -0,0 +1,18 @@
+# This file is a template generator for Natron's issue tracker dashboard
+
+# The blank template uses the old issue template
+blank_issues_enabled: true
+
+contact_links:
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/config.yml:6]: This is a good list! :D

File: .github/ISSUE_TEMPLATE/config.yml [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+  - name: Read the Code of Conduct
+    url: https://github.com/NatronGitHub/Natron/blob/RB-2.4/CODE_OF_CONDUCT.md
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/config.yml:8]: Links that will break :(

File: .github/ISSUE_TEMPLATE/feature.yml [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+            A feature is realistically implementable if you can reasonably expect yourself to implement the feature within 1 month of the request. A feature can still be added if it is not realistically implementable, but it will be tagged as an *idea* and relegated to be added in the indefinite future.
+        options:
+            - label: My feature is realistically implementable
+            - label: My feature is NOT realistically implementable
+  # Contributable or not?
+  - type: checkboxes
+    attributes:
+        label: Can you contribute in creating this feature?
+        options:
+            - label: I am unable, incapable, or not willing to contribute in creating this feature
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/feature.yml]: Unable encompasses incapable.  Minor grammar fix.
```suggestion
            - label: I am unable, or not willing to contribute to creating this feature
```

File: .github/ISSUE_TEMPLATE/feature.yml [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+            - label: I am able and willing to contribute in creating this feature
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/feature.yml]:
```suggestion
            - label: I am able and willing to contribute to the creation of this feature
```

File: .github/ISSUE_TEMPLATE/feature.yml [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+            - label: This feature is not on the [issue tracker](https://github.com/NatronGitHub/Natron/issues?q=is%3Abug+type%3Afeature+type%3Aidea) and is not a duplicate
+              required: true
+  # Feature request description
+  - type: textarea
+    attributes:
+        label: Provide a description of your feature request
+        render: markdown
+    validations:
+        required: true
+  # Realistically implementable or not?
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/feature.yml:27]: Would be interested in how useful @devernay finds this section.

File: .github/ISSUE_TEMPLATE/feature.yml [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+        label: Can you contribute in creating this feature?
+        options:
+            - label: I am unable, incapable, or not willing to contribute in creating this feature
+            - label: I am able and willing to contribute in creating this feature
+  # Additional details
+  - type: textarea
+    attributes:
+        label: Additional details
+        description: |
+          You do not have to fill this out but more details are always helpful.
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/feature.yml]: Encourage people to add them! Forms are allowed to be expressive!

```suggestion
          You do not have to fill this out but more details are always helpful!
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 17]
```diff
@@ -1,6 +1,33 @@
-Please read the [contribution guidelines](https://github.com/NatronGitHub/Natron/blob/master/CONTRIBUTING.md).
+Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. Additionally, make sure you've done all of these things:
 
-## Description
+- [ ] I've followed the [contributing guidelines](https://github.com/NatronGitHub/Natron/blob/RB-2.4/CODE_OF_CONDUCT.md) to the best of my understanding
```
Comment (Shrinks99) [inline .github/PULL_REQUEST_TEMPLATE.md:3]: More links that will break in the future :(

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 18]
```diff
@@ -1,6 +1,33 @@
...
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Improvement (non-breaking change which does not add functionality nor fixes a bug but improves Natron in some way)
+- [ ] Breaking change (fix or feature that would cause existing functionality to change)
+- [ ] My change requires a change to the documentation
+    - [ ] I have updated the documentation accordingly
+
+**Explain the motivation for making this change. What does the pull request do?**
+
+[Your answer]
```
Comment (Shrinks99) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
**What does this pull request do?**

[Your answer, explain the motivation for making this change.]
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 19]
```diff
@@ -1,6 +1,33 @@
...
+
+**Show a few screenshots (if this is a visual change)**
+
+[Your answer]
```
Comment (Shrinks99) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
[Your answer, delete this section if it is not a visual change.]
```

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 20]
```diff
@@ -0,0 +1,123 @@
...
+      value: |
+        Thank you for reporting a bug! Please first check the boxes below to ensure that your bug report meets our requirements.
+  # Requirements checks
+  - type: checkboxes
+    attributes:
+      options:
+        - label: I'm using the latest version of Natron (not required but recommended)
+        - label: I've restarted Natron and the issue persists
+          required: true
+        - label: I've run Natron via the [command line](https://natron.readthedocs.io/en/rb-2.4/devel/natronexecution.html) and the issue persists
```
Comment (Shrinks99) [inline .github/ISSUE_TEMPLATE/bug.yml:18]: Links that will break :(

**Review by Shrinks99 — CHANGES_REQUESTED** [review]
This looks like a positive change to how issues are created!  I think it will result in a better issue creation experience for users and more standardized & relevant information for developers.  I like it!

I have noted some points with links that will break every time we version up because we don't work out of the "master" or "main" branch (a name that doesn't change).  Bummer!  Changing this is outside of the scope of this PR but I hope it's considered at some point because it makes this kind of stuff annoying.

Once these changes are made I'm good with it I think!  Getting good answers from forms is hard and I think you've done a good job at it!

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 1 above)
Comment (Songtech-0912) [reply .github/ISSUE_TEMPLATE/bug.yml]: Hmm...good point. Added your suggested changes.

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 4 above)
Comment (Songtech-0912) [reply .github/ISSUE_TEMPLATE/bug.yml:70]: Added your suggestion (flatpak and snap both) as separate commit

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 17 above)
Comment (Songtech-0912) [reply .github/PULL_REQUEST_TEMPLATE.md:3]: Leave it for now? Until the new website goes up (where presumably we will put the contributing guidelines and code of conduct) we have no real other option.

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 20 above)
Comment (Songtech-0912) [reply .github/ISSUE_TEMPLATE/bug.yml:18]: Yeah, same deal as before - unless we build a url redirector into the website itself this isn't going to get solved, but I think it's bearable for now.

File: .github/ISSUE_TEMPLATE/feature.yml (hunk 15 above)
Comment (Songtech-0912) [reply .github/ISSUE_TEMPLATE/feature.yml:27]: Yeah - my original idea was to use this to categorize issues as either "features" or "ideas" (we have those two tags already). Not a must-have though.

**Review by Shrinks99 — APPROVED** [review]

**Pull request merged**